zerver/lib/avatar_hash.py

```diff
@@ -15,25 +15,26 @@ def gravatar_hash(email: str) -> str:
     return hashlib.md5(email.lower().encode()).hexdigest()
 
 
-def user_avatar_hash(uid: str) -> str:
+def user_avatar_hash(uid: str, version: str) -> str:
     # WARNING: If this method is changed, you may need to do a migration
     # similar to zerver/migrations/0060_move_avatars_to_be_uid_based.py .
 
-    # The salt probably doesn't serve any purpose now.  In the past we
-    # used a hash of the email address, not the user ID, and we salted
-    # it in order to make the hashing scheme different from Gravatar's.
-    user_key = uid + settings.AVATAR_SALT
+    # The salt prevents unauthenticated clients from enumerating the
+    # avatars of all users.
+    user_key = uid + ":" + version + ":" + settings.AVATAR_SALT
     return hashlib.sha1(user_key.encode()).hexdigest()
```

Comment:
Do we want to take the opportunity of rebuilding all the avatars to upgrade from `sha1` for this?  @andersk FYI.

Reply:
I guess there's no reason to _not_ move to SHA256.  Will update.

Reply:
Changed this to be a truncated SHA-256, so we're not doubling the bytes it takes to send avatar images.

Reply:
Looks like you forgot to update the migration on this detail, I'll take care of it.